api/src/lib.rs

```diff
@@ -137,6 +143,35 @@ pub use crate::{
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Token(pub String);
 
+pub trait SplittableRequest {
+    fn split(self) -> Vec<Self>
```

Comment:
Also, you should be able to define `split` as returning an `impl Iterator` instead of a `Vec` and remove the `collect()` at the end of the implementations.